packages/wonder-blocks-badge/src/components/badge.tsx

```diff
@@ -1,5 +1,202 @@
+        case "success":
+            return styles.successIcon;
+        case "warning":
+            return styles.warningIcon;
+        case "critical":
+            return styles.criticalIcon;
+        default:
+            return {};
+    }
+}
+
+/**
+ * Badges are visual indicators used to display concise information, such as
+ * a status, label, or count.
+ */
+export const Badge = (props: Props) => {
+    const {kind = "info", icon, label, ...otherProps} = props;
+    if (!label && !icon) {
+        return <React.Fragment />;
+    }
+    return (
+        <StyledDiv
+            style={[
+                typographyStyles.LabelXSmall,
+                styles.badge,
+                getKindStyle(kind),
+                icon && !label ? styles.iconOnly : {},
+            ]}
+            {...otherProps}
+        >
+            {icon && (
+                <StyledSpan style={[styles.icon, getIconKindStyle(kind)]}>
+                    {icon}
+                </StyledSpan>
+            )}
+            {label}
+        </StyledDiv>
+    );
+};
+
+const badgeTokens = {
+    root: {
+        sizing: {
+            paddingBlock: sizing.size_040,
+            paddingInline: sizing.size_080,
+            gap: sizing.size_040,
+        },
+        border: {
+            width: border.width.thin,
+            style: "solid",
+            radius: border.radius.radius_080,
+        },
+        color: {
+            info: {
+                background: semanticColor.status.notice.background,
+                border: semanticColor.status.notice.foreground,
+                foreground: semanticColor.text.primary,
+            },
+            success: {
+                background: semanticColor.status.success.background,
+                border: semanticColor.status.success.foreground,
+                foreground: semanticColor.text.primary,
+            },
+            warning: {
+                background: semanticColor.status.warning.background,
+                border: semanticColor.status.warning.foreground,
+                foreground: semanticColor.text.primary,
+            },
+            critical: {
+                background: semanticColor.status.critical.background,
+                border: semanticColor.status.critical.foreground,
+                foreground: semanticColor.text.primary,
+            },
+        },
+    },
+    icon: {
+        sizing: {
+            width: sizing.size_160,
+            height: sizing.size_160,
+        },
+        color: {
+            info: {
+                foreground: semanticColor.status.notice.foreground,
+            },
+            success: {
+                foreground: semanticColor.status.success.foreground,
+            },
+            warning: {
+                foreground: semanticColor.status.warning.foreground,
+            },
+            critical: {
+                foreground: semanticColor.status.critical.foreground,
+            },
+        },
+    },
+    iconOnly: {
```

Comment:
`iconOnly` styles... I'm not sure about where to place the padding token! For context, the badge has smaller padding when it is using only an icon. It's more of a variant of the `root` element, but I'm not sure how to structure it. I'm open to changing this!